vnext/Desktop/Modules/BlobModule.cpp

```diff
@@ -0,0 +1,239 @@
+
+using namespace facebook::xplat;
+
+using folly::dynamic;
+using std::lock_guard;
+using std::mutex;
+using std::string;
+using std::vector;
+using std::weak_ptr;
+using winrt::Windows::Foundation::GuidHelper;
+using winrt::Windows::Security::Cryptography::CryptographicBuffer;
+
+namespace {
+constexpr char moduleName[] = "BlobModule";
+constexpr char blobURIScheme[] = "blob";
+
+weak_ptr<Microsoft::React::IWebSocketModuleContentHandler> s_contentHandler;
+} // namespace
+
+namespace Microsoft::React {
+
+BlobModule::BlobModule() noexcept {
+  m_contentHandler = std::make_shared<BlobWebSocketModuleContentHandler>();
+  s_contentHandler = m_contentHandler;
+}
+
+#pragma region CxxModule overrides
+
+string BlobModule::getName() {
+  return moduleName;
+}
+
+std::map<string, dynamic> BlobModule::getConstants() {
+  return {{"BLOB_URI_SCHEME", blobURIScheme}, {"BLOB_URI_HOST", {}}};
+}
+
+// clang-format off
+std::vector<module::CxxModule::Method> BlobModule::getMethods() {
+  return
+  {
+    Method(
+      "addNetworkingHandler",
+      [this](dynamic args)
+      {
+        //TODO: Implement
+      }
+    ),
+
+    Method(
+      "addWebSocketHandler",
+      [this](dynamic args)
+      {
+        auto id = jsArgAsInt(args, 0);
+
+        m_contentHandler->Register(id);
+      }
+    ),
+
+    Method(
+      "removeWebSocketHandler",
+      [this](dynamic args)
+      {
+        auto id = jsArgAsInt(args, 0);
+
+        m_contentHandler->Unregister(id);
+      }
+    ),
+
+    Method(
+      "sendOverSocket",
+      [this](dynamic args)
+      {
+        auto blob = jsArgAsObject(args, 0);
+        auto blobId = blob["blobId"].getString();
+        auto offset = blob["offset"].getInt();
+        auto size = blob["size"].getInt();
+        auto socketID = jsArgAsInt(args, 1);
+
+        auto data = m_contentHandler->ResolveMessage(std::move(blobId), offset, size);
+
+        if (auto instance = getInstance().lock())
+        {
+          auto buffer = CryptographicBuffer::CreateFromByteArray(data);
+          auto winrtString = CryptographicBuffer::EncodeToBase64String(std::move(buffer));
+          auto base64String = Common::Unicode::Utf16ToUtf8(std::move(winrtString));
+
+          auto sendArgs = dynamic::array(std::move(base64String), socketID);
+          instance->callJSFunction("WebSocketModule", "sendBinary", std::move(sendArgs));
+        }
+      }
+    ),
+
+    Method(
+      "createFromParts",
+      [this](dynamic args)
+      {
+        auto parts = jsArgAsArray(args, 0);   // Array<Object>
+        auto blobId = jsArgAsString(args, 1);
+        vector<uint8_t> buffer{};
+
+        for(auto& part : parts)
+        {
+          auto type = part["type"];
+          if (type == "blob")
+          {
+            auto blob = part["data"];
+            auto blobId = blob["blobId"].asString();
+
+            auto bufferPart = m_contentHandler->ResolveMessage(blob["blobId"].asString(), blob["offset"].asInt(), blob["size"].asInt());
```

Comment:
>blob["blobId"].asString() [](start = 63, length = 25)

can we reuse `blobId` local variable?

Reply:
I think not. I's std::move'd in line 143.